### PR TITLE
Added support for avro and json schema converters for Source Connector

### LINKED ISSUE #193: Sink connector needs to be able to support Converters
Sink connector needs to be able to support Converters so that we can support JSON (string without a schema) as we do today, but also JSON with a schema, and AVRO. 

Add a test to ensure that when the connector is configured with avro and json-schema converters it works as expected.

### LINKED ISSUE #194: Source connector needs to be able to support Converters
Source connector needs to be able to support Converters so that we can support JSON (string without a schema) as we do today, but also JSON with a schema, and AVRO.

Add a test to ensure that when the connector is configured with avro and json-schema converters it works as expected.

### PR DESCRIPTION
**Pull request by kev-ms**
## Type of PR

- [ ] Documentation changes
- [x] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR
- Added JSON to Schema and Value conversions to support Json with schema conversion
- Added support for schema registry for AVRO conversion

## Review notes
- Validate Source Connector with below configurations:
    >"key.converter.schemas.enable": "true"
    >"value.converter.schemas.enable": "true"
    >"key.converter.class": "org.apache.kafka.connect.json.JsonConverter"
    >"value.converter.class": "org.apache.kafka.connect.json.JsonConverter"
- Validate io.confluent.connect.avro.AvroConverter by seeting below values
    >"key.converter.class": "io.confluent.connect.avro.AvroConverter"
    >"value.converter": "io.confluent.connect.avro.AvroConverter"
    >"key.converter.schema.registry.url": "http://schema-registry:8081"
    >"value.converter.schema.registry.url": "http://schema-registry:8081"
    >"key.converter.schemas.enable": "true"
    >"value.converter.schemas.enable": "true"

## Issues Closed or Referenced

- Closes #194 
- References #14 #160

### CODE REVIEW

File: src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/source/JsonToStruct.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import static org.apache.kafka.connect.data.Values.convertToDouble;
+import static org.apache.kafka.connect.data.Values.convertToFloat;
+import static org.apache.kafka.connect.data.Values.convertToInteger;
+import static org.apache.kafka.connect.data.Values.convertToLong;
+import static org.apache.kafka.connect.data.Values.convertToShort;
+
+import java.util.Iterator;
+import java.util.Map;
+
+public class JsonToStruct {
```
Comment (sivamu) [inline src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/source/JsonToStruct.java:22]: Sonarlint is reporting linter issues on line 30 and the lines with `logger.error`.

File: src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/source/JsonToStruct.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            case NUMBER:
+                if (jsonValue.isIntegralNumber()) {
+                    // Add support for INT8 & INT64
+                    return Schema.INT64_SCHEMA;
+                }
+                else {
+                    return Schema.FLOAT64_SCHEMA;
+                }
+            case ARRAY:
+                SchemaBuilder arrayBuilder = SchemaBuilder.array(jsonValue.elements().hasNext() ? inferSchema(jsonValue.elements().next()) : Schema.OPTIONAL_STRING_SCHEMA);
```
Comment (sivamu) [inline src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/source/JsonToStruct.java:65]: Nit but it would be nice to break up this line if possible

**Review by sivamu — COMMENTED** [review]
Minor changes needed.

Tested out the converter and it works very well, nice job! Got data in json, jsonschema and avro too. As a result, was able to use SMTs with the source connector outputs.

File: src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/source/JsonToStruct.java (hunk 1 above)
Comment (sivamu) [reply src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/source/JsonToStruct.java:22]: Didn't check earlier but sonarlint is finding issues with the fillrecords method in the sourcetask class

**Review by ryancrawcour — APPROVED** [review]

**Comment by kev-ms** [comment]
Closing this PR since #230 covers these changes

**Pull request closed without merging**